### PR TITLE
Platform service creating resource quotas by default

### PR DESCRIPTION
**Pull request by chrusty**
This gets the platform service to create a new resource-quota whenever it creates a namespace (in the same way that it creates a network-policy).

Not entirely sure how to handle the defaults... currently hard-coded, but this could be better. @domwong any thoughts?

### CODE REVIEW

**Comment by domwong** [comment]
I think as a first pass we should just make it config driven. We'll obvs need different quotas depending on platform vs dev tier. At some point we'll think about allowing users to pay for extra quota so we'll have a quota service but for now config is fine.

**Comment by chrusty** [comment]
> I think as a first pass we should just make it config driven. We'll obvs need different quotas depending on platform vs dev tier. At some point we'll think about allowing users to pay for extra quota so we'll have a quota service but for now config is fine.

OK cool, well I've added a commit which makes this thing use config. Regarding the "paying for extra quota" stuff, we should theoretically support Update() calls for ResourceQuotas.

File: platform/handler/platform.go [hunk 1]
```diff
@@ -33,6 +26,31 @@ type Platform struct {
 
 // New returns an initialised platform handler
 func New(service *service.Service) *Platform {
+
+	if val, err := config.Get("micro.platform.resource_limits.cpu"); err != nil {
+		defaultResourceLimits.CPU = int32(val.Int(8000))
+	}
+
+	if val, err := config.Get("micro.platform.resource_limits.ephemeralstorage"); err != nil {
```
Comment (asim) [inline platform/handler/platform.go:34]: can we just call it disk in our config? otherwise looks good

**Comment by asim** [comment]
Happy with that, merge at will. 

Set limits 1 core, 1gb mem, 10gb disk in dev. Set 2 core, 2gb mem, 20gb disk in platform. We then reevaluate as we go.

**Comment by asim** [comment]
We also need to retroactively apply this to existing namespaces.

File: platform/handler/platform.go (hunk 1 above)
Comment (chrusty) [reply platform/handler/platform.go]: OK i changed it

**Comment by domwong** [comment]
Tests do pass but flaky

**Comment by asim** [comment]
Nice! So now we can set this stuff and I'll update the fair usage limits in the m3o.dev docs and we can adjust as we see fit.

```
Dev:
1 core, 1gb mem, 10gb disk

Platform:
2 core, 2gb mem, 20gb disk
```

Must back port to all namespaces except micro I guess also

**Comment by domwong** [comment]
Cool. I'll sort this out. Will either do it today or Monday

**Pull request merged**